### PR TITLE
Login router + local storage

### PR DESCRIPTION
**Pull request by JoaquinBattilana**
## Summary

- Added redux thunk for async dispatches
-Used local storage for persisting the session
-Used react router for simple routing between login and game

## Screenshots

login-router

## Trello Card

https://trello.com/c/tMgUbW8E/50-routing-i-router

### CODE REVIEW

File: react/tic-tac-toe/src/app/components/ConditionalRedirect/index.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { Redirect } from 'react-router-dom';
+import React from 'react';
+
+function ConditionalRedirect({ condition, first, second }) {
+  return <Redirect to={condition() ? first : second} />;
```
Comment (erikankodelia) [inline react/tic-tac-toe/src/app/components/ConditionalRedirect/index.js]: Why do you pass the function and not the value that return de fuction directly?

File: react/tic-tac-toe/src/app/utils/validate.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export const isLogged = () => window.localStorage.getItem('token');
+export const isNotLogged = () => !window.localStorage.getItem('token');
```
Comment (erikankodelia) [inline react/tic-tac-toe/src/app/utils/validate.js]: It is not necessary this function, you can use !isLogged

File: react/tic-tac-toe/src/config/api/index.js [hunk 3]
```diff
@@ -1,8 +1,12 @@
 import { create } from 'apisauce';
 
-const api = create ({
+const headers = {
+  token: window.localStorage.getItem('token')
+};
+const api = create({
```
Comment (erikankodelia) [inline react/tic-tac-toe/src/config/api/index.js:7]: Add an empty line before

**Review by erikankodelia — CHANGES_REQUESTED** [review]

**Review by gon-escalante — APPROVED** [review]

File: react/tic-tac-toe/src/app/screens/Login/components/LoginForm/layout.js [hunk 4]
```diff
@@ -1,14 +1,21 @@
...
+  WRONG_INPUT_TEXT
+} from '~constants/text';
 
-function LoginForm({ handleSubmit, invalid }) {
+function LoginForm({ handleSubmit, invalid, hasError }) {
   return (
-    <form onSubmit={handleSubmit} className={styles.loginForm} >
+    <form onSubmit={handleSubmit} className={styles.loginForm}>
       <h2 className={styles.formTitle}>{LOGIN_TITLE}</h2>
+      {hasError ? <h3 className={styles.inputError}>{WRONG_INPUT_TEXT}</h3> : null}
```
Comment (erikankodelia) [inline react/tic-tac-toe/src/app/screens/Login/components/LoginForm/layout.js]: It is not necessary the null, you can write hasError && <h3 ...

**Review by erikankodelia — APPROVED** [review]

**Pull request merged**